client/extensions/woocommerce/app/orders/orders-list.js

```diff
@@ -46,16 +47,28 @@ class Orders extends Component {
 	componentWillReceiveProps( newProps ) {
 		const hasAnythingChanged = (
 			newProps.currentPage !== this.props.currentPage ||
+			newProps.currentSearch !== this.props.currentSearch ||
 			newProps.currentStatus !== this.props.currentStatus ||
 			newProps.siteId !== this.props.siteId
 		);
-		if ( newProps.siteId && hasAnythingChanged ) {
-			const query = {
-				page: newProps.currentPage,
-				status: newProps.currentStatus,
-			};
-			this.props.fetchOrders( newProps.siteId, query );
+		if ( ! newProps.siteId || ! hasAnythingChanged ) {
+			return;
 		}
+
+		const query = {
+			page: newProps.currentPage,
+			search: newProps.currentSearch,
+			status: newProps.currentStatus,
+		};
+		if ( newProps.currentSearch !== this.props.currentSearch ) {
+			this.props.updateCurrentOrdersQuery( this.props.siteId, { page: 1 } );
+			query.page = 1;
+		} else if ( newProps.currentStatus !== this.props.currentStatus ) {
+			this.props.updateCurrentOrdersQuery( this.props.siteId, { page: 1, search: '' } );
+			query.page = 1;
+			query.search = '';
+		}
+		this.props.fetchOrders( newProps.siteId, query );
```

Comment:
I think these lines ^ need a re-think, since I need to reset some parts of the query based on what just changed, but I'm not sure what to do - any ideas would be welcome :)